src/main/java/com/rackspace/jenkins_nodepool/NodePoolNode.java

```diff
@@ -116,6 +151,13 @@ public void setInUse() throws Exception {
      */
     public void release() throws Exception {
         setState("used");
+        unlock();
```

Comment:
Nodes are unlocked when they are held, so this unlock fails when the node is eventually deleted.

```